File: katas/8 kyu/rock-paper-scissors/solution.py
import codewars_test as test
# -------------------------------- my solution ------------------------------- #
def rps(p1, p2):
    rule = {'rock' : 'paper' , 'paper' : 'scissors' , 'scissors' : 'rock'}
    if p1 == p2 :
      return 'Draw!'
    return 'Player 2 won!' if p2 == rule[p1] else 'Player 1 won!'
# ------------------------------ clever solution ----------------------------- #
def rps(p1, p2):
    hand = {'rock':0, 'paper':1, 'scissors':2}
    results = ['Draw!', 'Player 1 won!', 'Player 2 won!']
    return results[hand[p1] - hand[p2]]
# ----------------------------------- test ----------------------------------- #
@test.describe("rock paper scissors")
def basic_tests():
    @test.it("Player 1 wins")
    def player_1():
        test.assert_equals(rps('rock', 'scissors'), "Player 1 won!")
    @test.it("Player 1 wins")
    def player_1():
        test.assert_equals(rps('scissors', 'rock'), "Player 2 won!")
    @test.it("Draw")
    def draw():
        test.assert_equals(rps('rock', 'rock'), 'Draw!')

File: katas/8 kyu/rock-paper-scissors/solution.js
// rock-paper-scissors
/* ------------------------------- MY SOLUTION ------------------------------ */
const rules = {
  "rock" : "paper",
  "paper" : "scissors",
  "scissors" : "rock"
}
const rps = (p1, p2) => {
  if (rules[p1] === rules[p2]) return "Draw!" 
  return `Player ${rules[p1] === p2 ? 2:1} won!`
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
chai.config.truncateThreshold=0;

describe('rock paper scissors', function() {
  const getMsg = (n) => `Player ${n} won!`;

  it('player 1 win', function() {
    Test(rps('rock', 'scissors'), getMsg(1));
    Test(rps('scissors', 'paper'), getMsg(1));
    Test(rps('paper', 'rock'), getMsg(1));
  });

  it('player 2 win', function() {
    Test(rps('scissors', 'rock'), getMsg(2));
    Test(rps('paper', 'scissors'), getMsg(2));
    Test(rps('rock', 'paper'), getMsg(2));
  });

  it('draw', function() {
    Test(rps('rock', 'rock'), 'Draw!');
    Test(rps('scissors', 'scissors'), 'Draw!');
    Test(rps('paper', 'paper'), 'Draw!');
  });
});